drivers/dma/dma_sedi.c

```diff
@@ -0,0 +1,446 @@
+						   channel, config);
+		if (ret) {
+			goto ERR;
+		}
+		src_addr = block_config->source_address;
+		dst_addr = block_config->dest_address;
+
+		ret = sedi_dma_start_transfer(info->peripheral_id, channel,
+						src_addr, dst_addr, block_config->block_size);
+	} else {
+		LOG_ERR("MULTIPLE_BLOCK CONFIG is not set");
+		goto ERR;
+	}
+
+	if (ret != SEDI_DRIVER_OK) {
+		goto ERR;
+	}
+
+	return ret;
+
+ERR:
+	LOG_ERR("dma transfer failed");
+	return ret;
+}
+
+static int dma_sedi_stop(const struct device *dev, uint32_t channel)
+{
+	const struct dma_sedi_config_info *const info = DEV_CFG(dev);
+
+	LOG_DBG("stopping dma: %p, %d", dev, channel);
+	sedi_dma_abort_transfer(info->peripheral_id, channel);
+
+	return 0;
+}
+
+static const struct dma_driver_api dma_funcs = { .config = dma_sedi_chan_config,
+						 .start = dma_sedi_start,
+						 .stop = dma_sedi_stop,
+						 .reload = dma_sedi_reload,
+						 .get_status = NULL
+};
+
+static int dma_sedi_init(const struct device *dev)
+{
+	const struct dma_sedi_config_info *const config = DEV_CFG(dev);
+
+	config->irq_config();
+
+	return 0;
+}
+
+#ifdef CONFIG_PM_DEVICE
+
+static bool is_dma_busy(sedi_dma_t dev)
+{
+	sedi_dma_status_t chn_status;
+
+	for (int chn = 0; chn < DMA_CHANNEL_NUM; chn++) {
+		sedi_dma_get_status(dev, chn, &chn_status);
+		if (chn_status.busy == 1) {
+			return true;
+		}
+	}
+	return false;
+}
+
+static int dma_change_device_power(const struct device *dev,
+		enum pm_device_action action)
+{
+	struct dma_sedi_driver_data *const data = DEV_DATA(dev);
+	const struct dma_sedi_config_info *const info = DEV_CFG(dev);
+	sedi_dma_t dma_dev = info->peripheral_id;
+	int ret;
+
+	sedi_power_state_t state;
+
+	switch (action) {
+	case PM_DEVICE_ACTION_RESUME:
+		state = SEDI_POWER_FULL;
+		break;
+	case PM_DEVICE_ACTION_SUSPEND:
+		if (is_dma_busy(dma_dev)) {
+			return -EBUSY;
+		}
+		state = SEDI_POWER_SUSPEND;
+		break;
+
+	default:
+		return -ENOTSUP;
+	}
+
+	for (uint8_t chn = 0; chn < DMA_CHANNEL_NUM; chn++) {
+		ret = sedi_dma_set_power(dma_dev, chn, state);
+		if (ret != SEDI_DRIVER_OK) {
+			return -EIO;
+		}
+	}
+
+	return 0;
+}
+
+#endif
+
+#define DMA_DEVICE_INIT_SEDI(inst) \
+	static void dma_sedi_##inst##_irq_config(void);			\
+									\
+	static struct dma_sedi_driver_data dma_sedi_dev_data_##inst; \
+	static const struct dma_sedi_config_info dma_sedi_config_data_##inst = { \
+		.peripheral_id = DT_INST_PROP(inst, peripheral_id), \
+		.chn_num = DT_INST_PROP(inst, dma_channels), \
+		.irq_config = dma_sedi_##inst##_irq_config \
+	}; \
+	DEVICE_DT_DEFINE(DT_INST(inst, DT_DRV_COMPAT), &dma_sedi_init, \
+	      NULL, &dma_sedi_dev_data_##inst, &dma_sedi_config_data_##inst, PRE_KERNEL_2, \
+	      CONFIG_KERNEL_INIT_PRIORITY_DEVICE, (void *)&dma_funcs); \
+									\
+	static void dma_sedi_##inst##_irq_config(void)			\
+	{								\
+		IRQ_CONNECT(DT_INST_IRQN(inst),				\
+			    DT_INST_IRQ(inst, priority), dma_isr,	\
+			    (void *)inst,			\
```

Comment:
use DT_INST_PROP(inst, peripheral_id) instead. no guarantee that inst 0 is SEDI_DMA_0.

Reply:
Thanks for this comment, change it.